src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -534,10 +534,11 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     break;
                 case 'tagged':
-                    if (!$arg->getAttribute('tag')) {
+                    if (!$arg->hasAttribute('tag')) {
```

Comment:
should be reverted

Reply:
I'm not agree, the DOMElement getAttribute returns an empty string '' if the attribute is not found 
I think hasAttribute test is better here

Reply:
why?

Reply:
because we do not need to cast an empty string to a boolean...
Moreover, according to me, from a code reading point of view it's more "accurate" compared to what is tested

Reply:
I'm sorry but that's not technically correct - there is no cast here. What you describe as more accurate actually introduces a bug: when the attribute is defined to the empty, no exception would be thrown anymore. So no, that's just worse, the rest are illusions.